afkak/consumer.py

```diff
@@ -721,19 +721,7 @@ def _handle_fetch_error(self, failure):
         self._request_d = None
 
         if failure.check(OffsetOutOfRangeError):
-            if self.auto_offset_reset == OFFSET_EARLIEST:
-                offset_request = OffsetRequest(
-                    self.topic, self.partition, OFFSET_EARLIEST, 1)
-            elif self.auto_offset_reset == OFFSET_LATEST:
-                offset_request = OffsetRequest(
-                    self.topic, self.partition, OFFSET_LATEST, 1)
-            else:
-                self._start_d.errback(failure)
-                return
-            self._request_d = self.client.send_offset_request([offset_request])
-            d = self._request_d
-            d.addCallback(self._handle_fetch_response)
-            d.addErrback(self._handle_fetch_error)
+            self._fetch_offset = self.auto_offset_reset
```

Comment:
What happens when self.auto_offset_reset is None?  I'm guessing a failure in _do_fetch().
Ugh. Looks like my comment (earlier today) on that other PR wasn't right about the self.auto_offset_reset == None case, since in the case where the default self.request_retry_max_attempts of 0 is used, it'll still hang/retry forever.  So, we still needs this:
```
if self.auto_offset_reset == None:
    self._start_d.errback(failure)
    return
```
In the `if failure.check(OffsetOutOfRangeError):` block.